File: internal/content/testdata/bench/csharp/is-fibonacci-number.cs
public class Solution {
    public bool is_fibonacci(long n) {
        if (n <= 0) return false;
        long a = 1, b = 1;
        while (a < n) {
            long temp = a + b;
            a = b;
            b = temp;
        }
        return a == n;
    }
}

File: internal/content/testdata/bench/rust/is-fibonacci-number.rs
fn is_fibonacci(n: i64) -> bool {
    if n <= 0 {
        return false;
    }
    let mut a: i64 = 1;
    let mut b: i64 = 1;
    while a < n {
        let next = a + b;
        a = b;
        b = next;
    }
    a == n
}
